include/envoy/config/subscription_factory.h

```diff
@@ -19,12 +19,13 @@ class SubscriptionFactory {
    * @param scope stats scope for any stats tracked by the subscription.
    * @param callbacks the callbacks needed by all Subscription objects, to deliver config updates.
    *                  The callbacks must not result in the deletion of the Subscription object.
+   * @param cluster_index used to specify index of management server.
    * @return SubscriptionPtr subscription object corresponding for config and type_url.
    */
   virtual SubscriptionPtr
   subscriptionFromConfigSource(const envoy::config::core::v3::ConfigSource& config,
                                absl::string_view type_url, Stats::Scope& scope,
-                               SubscriptionCallbacks& callbacks) PURE;
+                               SubscriptionCallbacks& callbacks, int cluster_index) PURE;
```

Comment:
This seems like some kind of layering violation. Shouldn't need to plumb raw integers around abstract interfaces like the subscription factory.